crates/pcb/src/fmt.rs

```diff
@@ -0,0 +1,187 @@
+use anyhow::{Context, Result};
+use clap::Args;
+use log::debug;
+use pcb_buildifier::Buildifier;
+use pcb_ui::prelude::*;
+use pcb_zen::file_extensions;
+use std::collections::HashSet;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+#[derive(Args, Debug, Default, Clone)]
+#[command(about = "Format .zen and .star files using buildifier")]
+pub struct FmtArgs {
+    /// One or more .zen/.star files or directories containing such files (non-recursive) to format.
+    /// When omitted, all .zen/.star files in the current directory are formatted.
+    #[arg(value_name = "PATHS", value_hint = clap::ValueHint::AnyPath)]
+    pub paths: Vec<PathBuf>,
+
+    /// Check if files are formatted correctly without modifying them.
+    /// Exit with non-zero code if any file needs formatting.
+    #[arg(long)]
+    pub check: bool,
+
+    /// Show diffs instead of writing files
+    #[arg(long)]
+    pub diff: bool,
+}
+
+/// Format a single file using buildifier
+fn format_file(buildifier: &Buildifier, file_path: &Path, args: &FmtArgs) -> Result<bool> {
+    debug!("Formatting file: {}", file_path.display());
+
+    if args.check {
+        buildifier.check_file(file_path)
+    } else if args.diff {
+        let diff = buildifier.diff_file(file_path)?;
+        if !diff.is_empty() {
+            print!("{}", diff);
+        }
+        Ok(true)
+    } else {
+        buildifier.format_file(file_path)?;
+        Ok(true)
+    }
+}
+
+/// Collect .zen and .star files from the provided paths
+pub fn collect_files(paths: &[PathBuf]) -> Result<Vec<PathBuf>> {
+    let mut unique: HashSet<PathBuf> = HashSet::new();
+
+    if !paths.is_empty() {
+        // Collect files from the provided paths (non-recursive)
```

Comment:
should we do this recursively like "cargo fmt" does? or maybe better to match the behavior of build/, but it would be cool to just run this on the root of a repo

Reply:
I guess this should work like cargo fmt, so makes sense